src/codecoverage/bot/code_coverage_bot/codecov.py

```diff
@@ -144,7 +144,12 @@ def go_from_trigger_mozilla_central(self):
             token=secrets[secrets.COVERALLS_TOKEN]
         )
         logger.info('Report generated successfully')
-
+        json_output = json.loads(output)
+        file_list = [file['name'] for file in json_output['source_files']]
+        expected_extensions = ['.js', '.cpp']
+        for extension in expected_extensions:
+            assert any([file.endswith(extension) for file in file_list]), 'No {} file in the generated report'.format(
+                extension)
         logger.info('Upload changeset coverage data to Phabricator')
         phabricatorUploader = PhabricatorUploader(self.repo_dir, self.revision)
         phabricatorUploader.upload(json.loads(output))
```

Comment:
Reuse the value you loaded earlier here, so we don't need to parse the JSON string twice.

Reply:
This still needs to be changed.

Reply:
You mean loading the file names from the report directly in the assert statement without making a separate list, right?

Reply:
No I mean here you can simply reuse `report` instead of doing `json.loads(output)` again: `phabricatorUploader.upload(report)`.

Reply:
Okay. Thanks for the clarification.